Libraries/AppState/NativeAppState.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {|
+    initialAppState: string,
+  |};
+  +getCurrentAppState: (
+    success: (appState: {|app_state: string|}) => void,
+    failure: (error: Object) => void,
```

Comment:
```
failure: (error: Error) => void,
```

Reply:
Since I don't think we're instantiating an actual Error object, the best is probably an object type with the fields specified but I defer to @RSNara

Reply:
I checked and as far as I can tell we don't use Error in any of the specs we've created so far

Reply:
It looks like the native implementation of this method also never calls the error callback. We should just remove the error callback. It looks like we have three options:

1. Remove the argument from the error callback.
2. Remove the error callback.
3. Leave the argument in the error callback as an Object. 

Thoughts?

Reply:
Although it's a bit more work, I'd prefer that we clean up the native code and remove the error callback. But I'll leave that decision up to you @ericlewis.

Reply:
Yeah I agree, I think we do at least 2 and preferably also remove it from native 👍

Reply:
I think it's fine to keep the native side as is for this PR. You can send a follow up cleanup if you wish.